tests/components/element-selector-card/element-selector-card.spec.tsx

```diff
@@ -0,0 +1,49 @@
+import { fireEvent, render, screen } from "@testing-library/react";
+import "jest-canvas-mock";
+import { ElementSelectorCard } from "src/components/element-selector-card";
+import { EElementType } from "src/context-providers/builder";
+import { TestHelper } from "src/util/test-helper";
+
+describe("element-selector-panel", () => {
+    afterEach(() => {
+        jest.restoreAllMocks();
+        jest.resetAllMocks();
+    });
+    describe("onClick", () => {
+        it("should run the onClick function when the card is clicked on", () => {
+            renderComponent({
+                type: EElementType.EMAIL,
+                onClick: mockOnClick,
+            });
+            fireEvent.click(getElementSelectorCard() as HTMLElement);
+            expect(mockOnClick).toHaveBeenCalled();
+        });
+    });
+});
+
+// =============================================================================
+// HELPER FUNCTIONS
+// =============================================================================
+
+const renderComponent = (
```

Comment:
This is a simple component with no usage of the context. Hence there is no need to wrap it in the providers helper. Just pass the appropriate props to the component directly.